python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -213,3 +225,69 @@ def processBeforeAddingToModeler(self, algorithm, model):
                         values.append(ValueFromOutput(alg.modeler_name, out.name))
 
         algorithm.params[self.LAYERS] = values
+
+    def mappedNameToLayer(self, lyr, expression, layersDict, context):
+        '''Try to identify if a real layer is mapped in the expression with a symbolic name.'''
+
+        nameToMap = lyr.name()
+
+        # get last scope of the expressionContext because should be that related
+        # with mapped variables
+        # The scope name should be "algorithm_inputs"
+        expContextLastScope = context.expressionContext().lastScope()
+
+        # check for layers directly added in the expression
+        if (nameToMap + "@") in expression:
+            layersDict[nameToMap] = lyr
+
+        # check for the layers that are mapped as input in a model
+        #  to do this check in the latest scope all passed variables
+        # to look for a variable that is a layer or a string filename ç
+        # to a layer
+        varId = None
+        varDescription = None
+
+        for varName in expContextLastScope.variableNames():
+
+            layer = expContextLastScope.variable(varName)
+
+            if not isinstance(layer, str) and not isinstance(layer, QgsMapLayer):
+                continue
+
+            if isinstance(layer, QgsMapLayer) and nameToMap not in layer.source():
+                continue
+
+            varId = varName
+            varDescription = expContextLastScope.description(varName)
+
+            # because there can be variable with None or "" description
+            # then skip them
+            if not varDescription:
+                continue
+
+            # check if it's description starts with Output as in:
+            #    Output 'Output' from algorithm 'calc1'
+            # as set in https://github.com/qgis/QGIS/blob/master/src/core/processing/models/qgsprocessingmodelalgorithm.cpp#L516
+            # but var in expression is called simply
+            #    'Output' from algorithm 'calc1'
+            elements = varDescription.split(" ")
+            if len(elements) > 1 and elements[0] == "Output":
```

Comment:
This looks very fragile. What about translations?

Reply:
Right about translatio! no idea in this moment how to infer the child algorithm that is producing "Output"... any suggestion where to see? I spent most of time to debug to find where to recover these information. But it's probably my lack of knowledge of some aspect of the architecture.

Reply:
@nyalldawson what if I parse using the split of QObject::tr( "Output '%1' from algorithm '%2'" ) ? eg. the same string as in https://github.com/qgis/QGIS/blob/master/src/core/processing/models/qgsprocessingmodelalgorithm.cpp#L516
Create a interdependence of c++ and python code with a translation string that it's not a good code... but the best code would be having child and variable definition inside the expression context and not only the string. but this is a refactoring and api change.